docs/root/start/sandboxes/tls-sni.rst

```diff
@@ -0,0 +1,176 @@
+.. _install_sandboxes_tls_sni:
+
+TLS Server name indication (``SNI``)
+====================================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+	Used to make ``HTTP`` requests.
+
+   :ref:`jq <start_sandboxes_setup_jq>`
+	Parse ``json`` output from the upstream echo servers.
+
+This example demonstrates an Envoy proxy that listens on multiple domains
+on the same ``IP`` address and provides separate ``TLS`` termination for each.
+
+It also demonstrates Envoy acting as a client proxy connecting to upstream
+``SNI`` services.
+
+.. _install_sandboxes_tls_sni_step1:
+
+Step 1: Create keypairs for each of the domain endpoints
+********************************************************
+
+Change directory to ``examples/tls-sni`` in the Envoy repository.
+
+The example creates three ``TLS`` endpoints and each will require their own
+keypairs.
+
+Create self-signed certificates for these endpoints as follows:
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/tls-sni
+
+   $ mkdir -p certs
+
+   $ openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 \
```

Comment:
I have a minor preference for using EC certs for new examles.

Reply:
OK sounds good and I agree, we should do this everywhere. @phlax do you want to do this in all examples in a follow up?

/wait-any

Reply:
yep lets do it in a follow up  #14048